schema.go

```diff
@@ -119,35 +119,35 @@ func findBuildByBuilder(db *mgo.Database, builder string, buildnum int) (*LogKee
 	return build, nil
 }
 
-func UpdateFailedTestsByBuildID(id interface{}) (int, error) {
+func UpdateFailedBuild(id interface{}) error {
 	if id == nil {
-		return 0, errors.New("no build id defined")
+		return errors.New("no build id defined")
 	}
 
 	db, closer := db.GetDatabase()
 	defer closer()
 
-	info, err := db.C(testsName).UpdateAll(bson.M{"build_id": id}, bson.M{"$set": bson.M{"failed": true}})
+	err := db.C(buildsName).UpdateId(id, bson.M{"$set": bson.M{"failed": true}})
 
-	return info.Updated, errors.Wrapf(err, "problem setting failed state on tasks %+v", info)
+	return errors.Wrapf(err, "problem setting failed state on build %v", id)
 }
 
-func GetOldTests(limit int) ([]Test, error) {
+func GetOldBuilds(limit int) ([]LogKeeperBuild, error) {
 	db, closer := db.GetDatabase()
 	defer closer()
-	query := getOldTestQuery()
+	query := getOldBuildQuery()
 
 	db.Session.SetSocketTimeout(2 * AmboyInterval)
 
-	var tests []Test
-	err := db.C(testsName).Find(query).Limit(limit).All(&tests)
+	var builds []LogKeeperBuild
+	err := db.C(buildsName).Find(query).Limit(limit).All(&builds)
 	if err != nil {
-		return nil, errors.Wrap(err, "error finding tests")
+		return nil, errors.Wrap(err, "error finding builds")
 	}
-	return tests, err
+	return builds, err
 }
 
-func getOldTestQuery() bson.M {
+func getOldBuildQuery() bson.M {
 	return bson.M{
 		"started": bson.M{"$lte": time.Now().Add(-deletePassedTestCutoff)},
 		"failed":  false,
```

Comment:
I think we might need to make this a "doesn't exist or is false, as most builds won't have this field, and this is, therefore weird.)

Reply:
How do I go about making sure this is indexed?

Reply:
Won't failed default to false? This is the same assumption we made when we were searching by test.

Reply:
Here are the indexes
```
mongos> db.builds.getIndexes()
[
	{
		"v" : 1,
		"key" : {
			"_id" : 1
		},
		"name" : "_id_",
		"ns" : "buildlogs.builds"
	},
	{
		"v" : 1,
		"key" : {
			"started" : 1
		},
		"name" : "started_1",
		"ns" : "buildlogs.builds",
		"expireAfterSeconds" : NumberLong(31536000)
	},
	{
		"v" : 1,
		"key" : {
			"buildnum" : 1,
			"builder" : 1
		},
		"name" : "buildnum_1_builder_1",
		"ns" : "buildlogs.builds"
	}
]
mongos> db.logs.getIndexes()
[
	{
		"v" : 1,
		"key" : {
			"_id" : 1
		},
		"name" : "_id_",
		"ns" : "buildlogs.logs"
	},
	{
		"v" : 1,
		"key" : {
			"build_id" : 1,
			"test_id" : 1,
			"seq" : 1
		},
		"name" : "build_id_1_test_id_1_seq_1",
		"ns" : "buildlogs.logs"
	},
	{
		"v" : 1,
		"key" : {
			"started" : 1
		},
		"name" : "started_1",
		"ns" : "buildlogs.logs",
		"expireAfterSeconds" : NumberLong(31536000)
	},
	{
		"v" : 1,
		"key" : {
			"build_id" : 1,
			"started" : 1
		},
		"name" : "build_id_1_started_1",
		"ns" : "buildlogs.logs",
		"background" : true
	}
]
mongos> db.tests.getIndexes()
[
	{
		"v" : 1,
		"key" : {
			"_id" : 1
		},
		"name" : "_id_",
		"ns" : "buildlogs.tests"
	},
	{
		"v" : 1,
		"key" : {
			"started" : 1
		},
		"name" : "started_1",
		"ns" : "buildlogs.tests",
		"expireAfterSeconds" : NumberLong(31536000)
	},
	{
		"v" : 1,
		"key" : {
			"build_id" : 1,
			"started" : 1
		},
		"name" : "build_id_1_started_1",
		"ns" : "buildlogs.tests"
	}
]
```